hwcomposer/wayland-hwc.cpp

```diff
@@ -556,70 +464,61 @@ create_window(struct display *display, bool use_subsurfaces, std::string appID,
         window->xdg_surface =
                 xdg_wm_base_get_xdg_surface(display->wm_base, window->surface);
         assert(window->xdg_surface);
-        
         xdg_surface_add_listener(window->xdg_surface,
-                                     &xdg_surface_listener, window);
+                                     &xdg_surface_listener, window.get());
 
         window->xdg_toplevel = xdg_surface_get_toplevel(window->xdg_surface);
         assert(window->xdg_toplevel);
-        xdg_toplevel_add_listener(window->xdg_toplevel, &xdg_toplevel_listener, window);
-        if (display->isMaximized || !display->height || !display->width)
-            xdg_toplevel_set_maximized(window->xdg_toplevel);
-        const hidl_string appID_hidl(appID);
-        hidl_string appName_hidl(appID);
-        if (appID != "Waydroid" && display->task)
-            display->task->getAppName(appID_hidl, [&](const hidl_string &value)
-                                      { xdg_toplevel_set_title(window->xdg_toplevel, value.c_str()); });
-        else
-            xdg_toplevel_set_title(window->xdg_toplevel, appID.c_str());
-
-        if (appID != "Waydroid")
-            appID = "waydroid." + appID;
-        xdg_toplevel_set_app_id(window->xdg_toplevel, appID.c_str());
+        xdg_toplevel_add_listener(window->xdg_toplevel, &xdg_toplevel_listener, window.get());
     } else if (display->shell) {
         window->shell_surface =
             wl_shell_get_shell_surface(display->shell, window->surface);
         assert(window->shell_surface);
-
-        wl_shell_surface_add_listener(window->shell_surface, &shell_surface_listener, window);
+        wl_shell_surface_add_listener(window->shell_surface, &shell_surface_listener, window.get());
         wl_shell_surface_set_toplevel(window->shell_surface);
-        if (display->isMaximized || !display->height || !display->width)
-            wl_shell_surface_set_maximized(window->shell_surface, display->output);
-        const hidl_string appID_hidl(appID);
-        hidl_string appName_hidl(appID);
-        if (appID != "Waydroid" && display->task)
-            display->task->getAppName(appID_hidl, [&](const hidl_string &value)
-                                      { wl_shell_surface_set_title(window->shell_surface, value.c_str()); });
-        else
-            wl_shell_surface_set_title(window->shell_surface, appID.c_str());
     } else {
-        assert(0);
+        abort();
     }
 
+    if (display->isMaximized || calibrating) {
+        window->set_maximize(true);
+    }
+    // Set title
+    const hidl_string appID_hidl(appID);
+    if (window->appID != "Waydroid" && display->task) {
+        display->task->getAppName(appID_hidl, [&](const hidl_string &value){
+            window->set_title(value.c_str());
+        });
```

Comment:
```suggestion
    if (appID != "Waydroid" && display->task) {
        const hidl_string appID_hidl(appID);
        display->task->getAppName(appID_hidl, [&](const hidl_string &value){
            window->set_title(value.c_str());
        });
```

`window->appID` is unset here; I believe you meant to use the `appID` function argument.
Also `appID_hidl` can be defined inside the if scope.